core/src/main/java/com/microsoft/applicationinsights/LowLevelClient.java

```diff
@@ -0,0 +1,119 @@
+        }
+        // TODO check existing retry policy, and its configuration
+        this.pipeline = pipeline.build();
+        this.endpoint = endpoint;
+    }
+
+    CompletableResultCode send(List<TelemetryItem> telemetryItems) {
+        List<ByteBuffer> byteBuffers;
+        try {
+            byteBuffers = encode(telemetryItems);
+        } catch (Throwable t) {
+            logger.error("Error encoding telemetry items: {}", t.getMessage(), t);
+            return CompletableResultCode.ofFailure();
+        }
+        try {
+            return internalSend(byteBuffers);
+        } catch (Throwable t) {
+            logger.error("Error sending telemetry items: {}", t.getMessage(), t);
+            return CompletableResultCode.ofFailure();
+        }
+    }
+
+    List<ByteBuffer> encode(List<TelemetryItem> telemetryItems) throws IOException {
+        ByteBufferOutputStream out = new ByteBufferOutputStream(byteBufferPool);
+        for (Iterator<TelemetryItem> i = telemetryItems.iterator(); i.hasNext(); ) {
+            mapper.writeValue(out, i.next());
+            if (i.hasNext()) {
+                out.write('\n');
+            }
+        }
+        out.close(); // closing ByteBufferOutputStream is not really needed, but this makes LGTM happy
+        List<ByteBuffer> byteBuffers = out.getByteBuffers();
+        for (ByteBuffer bb : byteBuffers) {
+            bb.flip();
+        }
+        return byteBuffers;
+    }
+
+    private CompletableResultCode internalSend(List<ByteBuffer> byteBuffers) {
+        HttpRequest request = new HttpRequest(HttpMethod.POST, endpoint + "v2/track");
+
+        request.setBody(Flux.fromIterable(byteBuffers));
+
+        int contentLength = byteBuffers.stream().mapToInt(ByteBuffer::limit).sum();
+        request.setHeader("Content-Length", Integer.toString(contentLength));
+
+        // need to suppress the default User-Agent "ReactorNetty/dev", otherwise Breeze ingestion service will put that
+        // User-Agent header into the client_Browser field for all telemetry that doesn't explicitly set it's own
+        // UserAgent (ideally Breeze would only have this behavior for ingestion directly from browsers)
+        // TODO(trask)
+        //  not setting User-Agent header at all would be a better option, but haven't figured out how to do that yet
+        request.setHeader("User-Agent", "");
+
+        // TODO(trask) subscribe with listener
+        //  * retry on first failure (may not need to worry about this if retry policy in pipeline already, see above)
+        //  * write to disk on second failure
+        CompletableResultCode result = new CompletableResultCode();
+        pipeline.send(request)
+                .contextWrite(Context.of(Tracer.DISABLE_TRACING_KEY, true))
+                .subscribe(response -> {
+                    // TODO parse response, looking for throttling, partial successes, etc
+                    System.out.println("on response: " + response);
```

Comment:
can we use logger instead of sout?

Reply:
👍 commented it out, this is still work-in-progress